### PR TITLE
internal/ethapi: eth_getProof return empty hash instead of zero

### PR DESCRIPTION
**Pull request by jsvisa**
fix https://github.com/ethereum/go-ethereum/issues/28441

### CODE REVIEW

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -689,7 +689,13 @@ func (s *BlockChainAPI) GetProof(ctx context.Context, address common.Address, st
 		return nil, err
 	}
 	codeHash := statedb.GetCodeHash(address)
+	if codeHash == (common.Hash{}) {
+		codeHash = types.EmptyCodeHash
+	}
```
Comment (karalabe) [inline internal/ethapi/api.go:694]: This fixes this specific API call, but shouldn't this issue be fixed in statedb? Maybe it has some deeper implications?

Comment (jsvisa) [reply internal/ethapi/api.go]: thanks for the advice, now fix it in package statedb

**Review by holiman — CHANGES_REQUESTED** [review]
Yeah so this is 100% a consensus-breaking change, afaict, due to the reasons @rjl493456442 points out

**Comment by jsvisa** [comment]
So let's check it at the API's level, don't change in statedb's side.

**Comment by holiman** [comment]
Having looked a bit closer into what's being fixed here, I think the behaviour in https://github.com/ethereum/go-ethereum/issues/28441 is _wrong_, and therefore this PR is also not correct. It's IMO not correct to say that e.g. the codehash of an account which _does not at all exist in the trie_ has a certain `emptyCodeHash`.

**Pull request closed without merging**